app/main.py

```diff
@@ -1 +1,34 @@
-# write your code here
+
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.health = health
+        self.name = name
+        self.hidden = False
+        self.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, "
+                f"Hidden: {self.hidden}}}")
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        if self.hidden:
+            self.hidden = False
+        elif not self.hidden:
+            self.hidden = True
+
+
+class Carnivore(Animal):
+    def bite(self, obj: Animal) -> None:
+        if isinstance(obj, Herbivore) and not obj.hidden:
+            obj.health -= 50
+            if obj.health <= 0:
+                Animal.alive = [
+                    animal
+                    for animal in Animal.alive
+                    if obj.name != animal.name
```

Comment:
You don't need to use list comprehension here, you can use lists method `remove`